src/algebra/lie/of_associative.lean

```diff
@@ -66,34 +66,47 @@ lemma of_associative_ring_bracket (x y : A) : ⁅x, y⁆ = x*y - y*x := rfl
 
 end lie_ring
 
-namespace lie_algebra
+section lie_algebra
 
 variables {R : Type u} [comm_ring R] [algebra R A]
 
 /-- An associative algebra gives rise to a Lie algebra by taking the bracket to be the ring
 commutator. -/
 @[priority 100]
-instance of_associative_algebra : lie_algebra R A :=
+instance lie_algebra.of_associative_algebra : lie_algebra R A :=
 { lie_smul := λ t x y,
     by rw [lie_ring.of_associative_ring_bracket, lie_ring.of_associative_ring_bracket,
            algebra.mul_smul_comm, algebra.smul_mul_assoc, smul_sub], }
 
+namespace alg_hom
+
+variables {B : Type w} {C : Type w₁} [ring B] [ring C] [algebra R B] [algebra R C]
+variables (f : A →ₐ[R] B) (g : B →ₐ[R] C)
+
 /-- The map `of_associative_algebra` associating a Lie algebra to an associative algebra is
 functorial. -/
-def of_associative_algebra_hom {B : Type w} [ring B] [algebra R B] (f : A →ₐ[R] B) : A →ₗ⁅R⁆ B :=
+def to_lie_hom : A →ₗ⁅R⁆ B :=
```

Comment:
More future work: I think we should remove this in favor of `f.to_nonunital_alg_hom.to_lie_hom`, assuming I'm right in saying that definition does or can exist. You'll still be able to take advantage of transitive coercions.